File: py/features.py
# Thesmsworks SDK feature factory

from feature.base_feature import ThesmsworksBaseFeature
from feature.test_feature import ThesmsworksTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ThesmsworksBaseFeature(),
        "test": lambda: ThesmsworksTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()

File: elixir/lib/features.ex
# Thesmsworks SDK feature factory

defmodule Thesmsworks.Features do
  def make_feature(name) do
    case name do
      "debug" -> Thesmsworks.Feature.Debug.new()
      "idempotency" -> Thesmsworks.Feature.Idempotency.new()
      "metrics" -> Thesmsworks.Feature.Metrics.new()
      "paging" -> Thesmsworks.Feature.Paging.new()
      "ratelimit" -> Thesmsworks.Feature.Ratelimit.new()
      "retry" -> Thesmsworks.Feature.Retry.new()
      "test" -> Thesmsworks.Feature.Test.new()
      "timeout" -> Thesmsworks.Feature.Timeout.new()
      _ -> Thesmsworks.Feature.new()
    end
  end
end
